Library/Formula/mono.rb

```diff
@@ -30,7 +33,13 @@ def install
       --prefix=#{prefix}
       --enable-nls=no
     ]
-    args << "--build=" + (MacOS.prefer_64_bit? ? "x86_64": "i686") + "-apple-darwin"
+
+    if build.with? '64bit'
+      args << "--build=x86_64-apple-darwin"
+    else
+      # no option specified
+      args << "--build=" + (MacOS.prefer_64_bit? ? "x86_64": "i686") + "-apple-darwin"
```

Comment:
No docs on how to change the 'prefer 64 bit' flag, and besides that, you may need to have both installed anyhow. See Mono page on why they are shipping 32 bit by default - TLDR: they need it for some graphics lib which doesn't have 64 bit support.